tests/functional/test_no_color.py

```diff
@@ -0,0 +1,61 @@
+"""
+Test specific for the --no-color option
+"""
+import os
+import platform
+import subprocess as sp
+import sys
+
+import pytest
+
+
+@pytest.mark.skipif(sys.platform == 'win32',
+                    reason="does not run on windows")
+def test_no_color(script):
+
+    """
+    Test uninstalling an existing package - should out put red error
+
+    We must use subprocess with the script command, since redirection
+    in unix platform causes text coloring to disapper. Thus, we can't
+    use the testing infrastructure that other options has.
+    """
+
+    sp.Popen("script --flush --quiet --return /tmp/colored-output.txt"
+             " --command \"pip uninstall noSuchPackage\"", shell=True,
+             stdout=sp.PIPE, stderr=sp.PIPE).communicate()
+
+    with open("/tmp/colored-output.txt", "r") as result:
+
+        red_lines = 0
+        reset_lines = 0
+
+        for line in result.readlines():
```

Comment:
IMO, instead of all this, just doing `assert "\x1b" in result.read()` would be enough.

Reply:
Yes, you can only check that the red marker started, and ignore if it is not reset.
I thought someone would ask me to be a perfectionist here, so I implemented this way.

Reply:
Also, `\x1b` is just the color sequence begin. I would prefer to test that is really red, and not for example green or blue, for that uninstall case.

Reply:
> I would prefer to test that is really red, and not for example green or blue, for that uninstall case.

IMO, this test is meant only for testing that `--no-color` makes pip stop printing colour sequences. What that sequence is or if it's even valid is not related to that purpose of this test.